library/src/main/java/com/getstream/sdk/chat/utils/StringUtility.java

```diff
@@ -47,7 +43,39 @@ public static boolean isEmoji(String message) {
     }
 
     public static boolean containsMarkdown(String message) {
-        return Pattern.compile(MARKDOWN_PATTERN).matcher(message).find();
+        if (checkSymbolBrackets(message, "*") || checkSymbolBrackets(message, "_")
+                || checkSymbolBrackets(message, "#")
+                || checkSymbolBrackets(message, "**") || checkSymbolBrackets(message, "__")
+                || message.contains("~~") || message.contains("#{")
+                || message.contains("---") || message.contains("***") || message.contains("___")
+                || message.contains(">{")
+                || checkSymbolBrackets(message,"`") || checkSymbolBrackets(message, "```") || message.contains("//")
+                || message.contains("<i>") || message.contains("<em>") || message.contains("<cite>") || message.contains("<dfn>")
+                || message.contains("<b>") || message.contains("<strong>")
+                || message.contains("<sup>") || message.contains("<sub>")
+                || message.contains("<u>") || message.contains("<ins>")
+                || message.contains("<a>") || message.contains("<ul>") || message.contains("<ol>")
+                || message.contains("<img>") || message.contains("<blockquote>")
+                || message.contains("<h1>") || message.contains("<h2>") || message.contains("<h3>")
+                || message.contains("<h4>") || message.contains("<h5>") || message.contains("<h6>")
+                || message.startsWith("# ") || message.startsWith("## ") || message.startsWith("### ")
+                || message.startsWith("#### ") || message.startsWith("##### ")
+        ) {
+            return true;
+        }
+        return false;
+    }
+
+    public static boolean containsMention(String message) {
+        return message.contains("@");
+    }
+
+    public static boolean isMessageDeleted(Message message) {
+        return message.getType().equals("deleted");
```

Comment:
Sometimes it works wrong. maybe backend issue.
Please use `return message.getDeletedAt() != null;` instead.

Reply:
[Fixed](https://github.com/GetStream/stream-chat-android/pull/361/commits/1b71bd45a9b4fb981cdcbf9a0cc038068cd78b71
)